### PR TITLE
Bake the semantic argmax class map into the OpenVINO export graph

### PR DESCRIPTION
**Pull request by JESUSROYETH**
Semantic export bakes the per-pixel `argmax` into the graph and returns a compact `[B, H, W]` class map for ONNX, MNN, TensorRT>=10 and multi-class Hailo-10/15, but OpenVINO was left out of that set and still returns the full `[B, nc, H, W]` float logits. So an OpenVINO deployment runs the `argmax` on host and moves a tensor ~76x larger than needed.

OpenVINO supports integer graph outputs (`uint8`/`int32`), so it can join ONNX/MNN in the unconditional bake.

**Changes**

- Add `openvino` to the bake set in `SemanticSegment.forward`, and to the format lists in the inline comment and the `forward` docstring:

```python
if self.format in {"onnx", "mnn", "openvino"} or (self.format in {"engine", "hailo"} and self.bake_argmax):
```

- Add to `test_export_openvino_matrix` the same semantic class-map assertion that `test_export_onnx_matrix` already has, so the OpenVINO matrix checks the exported map end-to-end like the ONNX one.

Deleted: nothing — this adds OpenVINO to an existing format-gated branch that already implements the exact behaviour, so there is no path to delete or relocate; it only makes OpenVINO consistent with the other formats.

**Validation** (yolo26s-sem, i9-13900HX, OpenVINO 2026.2, CPU)

- Output shrinks from `[1, 19, 640, 640]` f32 (31.1 MB) to `[1, 640, 640]` uint8 (0.41 MB), a copy 76x smaller.
- End-to-end latency drops from 75.8 ms to 39.9 ms (**1.9x**): the in-graph reduction removes a 34 ms host `argmax`, and the graph alone is also ~2 ms faster because it writes the small map instead of the full logits. On GPU/iGPU the gain is larger — it is exactly the ~80x smaller device-to-host copy the comment already cites.
- The baked map is exact: on the same input, the in-graph `argmax` matches the host `argmax` of the logits in 409600/409600 pixels (0 mismatches), and it is byte-identical to what ONNX/MNN already emit.
- INT8 export still works: `quantize=8` produces the `uint8 [1, 640, 640]` map (6.4 MB), the reduction survives quantization.
- The exported IR changes as expected: on `main` it emits `[1, 19, 32, 32]` f32 logits (nano model, `imgsz=32`); with this change it emits the `[1, 32, 32]` uint8 class map, confirmed by reading the model outputs with `ov.Core().read_model`.
- `test_export_openvino_matrix` semantic combos pass with the new assertion, including `dynamic=True, quantize=8, batch=2` (NNCF INT8 calibration).
- Both consumers already handle the map: `SemanticSegmentationPredictor.postprocess` branches on `pred.ndim == 2` and `SemanticSegmentationValidator.postprocess` on `preds.ndim == 3` — the same paths the ONNX/MNN exports already use, so `predict` and `val` both work unchanged.

The baked class map is resized with nearest interpolation at postprocess (as it already happens for ONNX/MNN), so it differs from the raw-logits bilinear path by ~1% of pixels at object boundaries. That is the existing, intended tradeoff of baked semantic exports, now applied to OpenVINO for consistency, not a new behaviour.

Extends #25280.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Bake semantic segmentation argmax class maps directly into OpenVINO export graphs for compact integer outputs.

### 📊 Key Changes
- Adds `openvino` to the export formats that perform semantic class reduction in `SegmentationModel`.
- Returns `uint8` for models with up to 256 classes and `int32` for larger class counts.
- Extends OpenVINO semantic export tests to verify the mask exists and uses an expected integer dtype.
- Updates the module documentation to describe OpenVINO's compact output behavior.

### 🎯 Purpose & Impact
- Reduces exported semantic segmentation output size and host-device data transfer overhead.
- Provides consistent compact class-map behavior across OpenVINO, ONNX, MNN, and supported accelerator exports.
- Improves confidence in OpenVINO semantic inference through automated dtype and output validation. 🚀

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JESUSROYETH, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is._" — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. This review is automated, and an Ultralytics engineer will assist with any follow-up questions. Thank you for contributing to Ultralytics! 🚀

File: tests/test_exports.py [hunk 1]
```diff
@@ -292,7 +292,10 @@ def test_export_openvino_matrix(task, dynamic, quantize, batch, nms, end2end):
         nms=nms,
         end2end=end2end,
     )
-    YOLO(file)([SOURCE] * batch, imgsz=64 if dynamic else 32, batch=batch)  # exported model inference
+    r = YOLO(file)([SOURCE] * batch, imgsz=64 if dynamic else 32, batch=batch)  # exported model inference
+    if task == "semantic":
```
Comment (UltralyticsAssistant) [inline tests/test_exports.py]: 💡 **MEDIUM**: These assertions do not verify that the OpenVINO graph emits a baked class map. `SemanticSegmentationPredictor.postprocess()` accepts either logits or a `[H, W]` map and always converts logits through host-side `argmax` to an integer `semantic_mask`, so the pre-change `[B, nc, H, W]` OpenVINO output would pass unchanged. Inspect the raw `AutoBackend`/OpenVINO output and assert its rank or shape is `[B, H, W]` to cover the behavior this PR adds.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The OpenVINO bake change is placed in the semantic head’s existing export-format gate and is consistent with the ONNX/MNN path. One test issue remains: the added end-to-end assertions cannot distinguish baked outputs from the previous host-side argmax behavior.

💬 Posted 1 inline comment

**Comment by glenn-jocher** [comment]
The raw OpenVINO IR rank assertion now covers the baked class-map regression. Please also assert the output element type is `u8` or `i32` to lock down the compact integer-output contract.

**Comment by glenn-jocher** [comment]
Verified independently before merge on OpenVINO 2025.4.1. Reading the exported IR confirms the change: `[1,19,32,32] f32` logits on `main` -> `[1,32,32] uint8` class map with the patch. `torch2openvino` converts the TorchScript trace directly (not via ONNX), so the head genuinely sees `format='openvino'` — the addition is doing real work. All 8 `test_export_openvino_matrix` semantic combinations pass, including INT8/NNCF, `dynamic=True` and `batch=2`.

I removed the added test assertion (we keep tests out of `main` by policy). The head change and docstring are unchanged.

**Review by glenn-jocher — APPROVED** [review]
Verified locally: exported IR goes from `[1,19,32,32] f32` to `[1,32,32] uint8`, and all 8 semantic `test_export_openvino_matrix` combinations pass (incl. INT8/NNCF, dynamic, batch=2). Test removed per policy.
